### PR TITLE
Fendl update

### PR DESCRIPTION
**Pull request by SamPUG**
This pull request contains changes related to item 6  in our task list and issue 21 in the parent openmc-dev/data repository.

I have included the following in the convert_fendl script:

- Photon data for FENDL-3.1a
- Photon data for FENDL-3.1d
- Neutron and Photon data for FENDL-3.0 
- Neutron and Photon data for FENDL-2.1

I also attempted to add FENDL-2.0 but faced problems because one of the energy laws required was not supported. This was discussed in the comments for the GitHub issue and it was decided that this release could be left out for now. 

I had some issues integrating FENDL-3.0 as K-39 ace file contained 'Inf' values within the data. OpenMC is unable to deal with this and cannot convert the file to HDF5. Although I looked into generating the file using the NJOY functionality built into OpenMC, it was decided that it was best to exclude K-39. I have implemented a check and warning to test if the offending ace file is being processed. 

The FENDL-2.1 release is downloaded as individual ace files as it is not available in the same format as other releases. I thought it was best to create a download folder rather than download them in the current working directory.

Photo ENDF data for FENDL-2.1 is all within a single ENDF file. As such I added functionality which divides the file up into individual ENDF files. 

I have checked to make sure that the HDF5 files are generated for the added releases/particles and have checked the `cross_sections.xml` file contains the required libs. Let me know if you think any other testing is necessary.

### CODE REVIEW

File: convert_fendl.py [hunk 1]
```diff
@@ -41,90 +41,289 @@ class CustomFormatter(argparse.ArgumentDefaultsHelpFormatter,
...
+
+ace_files_dir = cwd.joinpath('-'.join([library_name, args.release, 'ace']))
+endf_files_dir = cwd.joinpath('-'.join([library_name, args.release, 'endf']))
+
+download_path = cwd.joinpath('-'.join([library_name, args.release, 'download']))
 # the destination is decided after the release is know to avoid putting the release in a folder with a misleading name
 if args.destination is None:
     args.destination = Path('-'.join([library_name, args.release, 'hdf5']))
 
 # This dictionary contains all the unique information about each release. This can be exstened to accommodated new releases
```
Comment (simondrichards) [inline convert_fendl.py]: Although this comment has not been changed in this PR, we could take the opportunity to fix the mangled English.

**Review by simondrichards — COMMENTED** [review]
Looks good. I've made one comment on an existing comment with mangled English which we might as well fix while we're in here.

My only other thought is whether we should clean up the download files after the data have been extracted. I assume they are no longer required and I don't see the sense of leaving the files lying around. Is it worth adding another argument to control whether the downloads are deleted after extracting the data? Personally I would think that the default should be to delete them.

**Comment by SamPUG** [comment]
Thanks for the review @simondrichards. I've added a `--cleanup`  and corresponding `--no-cleanup` argument which can be passed to remove the downloaded files. The `--no-cleanup` option is the default and its inclusion is slightly redundant but it follows the style of the other arguments. I chose to do `--no-cleanup` by default as this is the behavior of all the other convert scripts.

I also made one other minor change which was to change the position of the ENDF separation step for release 2.1. In it's previous position, it would have been attempted even if the `--no-extract` flag was given, potentially overwriting any changes made by a user. I think its position now makes more sense.

**Comment by simondrichards** [comment]
Thanks @SamPUG - all looks good to merge and submit a pull request to openmc-dev

**Pull request merged**